integration/mediation-tests/tests-mediator-2/src/test/java/org/wso2/carbon/esb/mediator/test/iterate/IterateJsonPathTest.java

```diff
@@ -0,0 +1,196 @@
+    public void testIterateMediatorFlowWithSequential() throws Exception {
+        URL endpoint = new URL(getApiInvocationURL("iteratesample8") + "/iteratejson8");
+        executeSequenceAndAssertResponse(endpoint);
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void stop() throws Exception {
+        super.cleanup();
+    }
+
+    private void executeSequenceAndAssertResponse(URL endpoint) throws Exception {
+
+        Map<String, String> header = new HashMap<String, String>();
+        header.put("Content-Type", "application/json");
+
+        HttpResponse httpResponse = doPost(endpoint, input, header);
+
+        assertTrue(httpResponse.getData().contains("Alice"), "Required element not found in aggregrated payload");
+        assertTrue(httpResponse.getData().contains("Bob"), "Required element not found in aggregrated payload");
+        assertTrue(httpResponse.getData().contains("Camry"), "Required element not found in aggregrated payload");
+    }
+
+    private static HttpResponse doPost(URL endpoint, String postBody, Map<String, String> headers)
+            throws AutomationFrameworkException, IOException {
+        HttpURLConnection urlConnection = null;
+        try {
+            urlConnection = (HttpURLConnection) endpoint.openConnection();
+            try {
+                urlConnection.setRequestMethod("POST");
+            } catch (ProtocolException e) {
+                throw new AutomationFrameworkException("Shouldn't happen: HttpURLConnection doesn't support POST?? " +
+                        e.getMessage(), e);
+            }
+            urlConnection.setDoOutput(true);
+            urlConnection.setDoInput(true);
+            urlConnection.setUseCaches(false);
+            urlConnection.setAllowUserInteraction(false);
+            urlConnection.setReadTimeout(10000);
+            for (Map.Entry<String, String> e : headers.entrySet()) {
+                urlConnection.setRequestProperty(e.getKey(), e.getValue());
+            }
+            OutputStream out = urlConnection.getOutputStream();
+            Writer writer = null;
+            try {
+                writer = new OutputStreamWriter(out, "UTF-8");
+                writer.write(postBody);
+            } catch (IOException e) {
+                throw new AutomationFrameworkException("IOException while posting data " + e.getMessage(), e);
+            } finally {
+                if (writer != null) {
+                    writer.close();
+                }
+            }
+            StringBuilder sb = new StringBuilder();
+            BufferedReader rd = null;
+            try {
+                rd = new BufferedReader(new InputStreamReader(urlConnection.getInputStream(), Charset.defaultCharset()));
+                String line;
+                while ((line = rd.readLine()) != null) {
+                    sb.append(line);
+                }
+            } finally {
+                if (rd != null) {
```

Comment:
Try will resources can be used for this puropose

Reply:
Fixed in https://github.com/wso2/product-ei/pull/4782